packages/core/src/telemetry/metrics.ts

```diff
@@ -6,38 +6,305 @@
-  METRIC_STARTUP_TIME,
-  METRIC_MEMORY_USAGE,
-  METRIC_CPU_USAGE,
-  METRIC_TOOL_QUEUE_DEPTH,
-  METRIC_TOOL_EXECUTION_BREAKDOWN,
-  METRIC_TOKEN_EFFICIENCY,
-  METRIC_API_REQUEST_BREAKDOWN,
-  METRIC_PERFORMANCE_SCORE,
-  METRIC_REGRESSION_DETECTION,
-  METRIC_REGRESSION_PERCENTAGE_CHANGE,
-  METRIC_BASELINE_COMPARISON,
-} from './constants.js';
+import { SERVICE_NAME, EVENT_CHAT_COMPRESSION } from './constants.js';
 import type { Config } from '../config/config.js';
 import type { ModelRoutingEvent, ModelSlashCommandEvent } from './types.js';
 
+const TOOL_CALL_COUNT = 'gemini_cli.tool.call.count';
+const TOOL_CALL_LATENCY = 'gemini_cli.tool.call.latency';
+const API_REQUEST_COUNT = 'gemini_cli.api.request.count';
+const API_REQUEST_LATENCY = 'gemini_cli.api.request.latency';
+const TOKEN_USAGE = 'gemini_cli.token.usage';
+const SESSION_COUNT = 'gemini_cli.session.count';
+const FILE_OPERATION_COUNT = 'gemini_cli.file.operation.count';
+const INVALID_CHUNK_COUNT = 'gemini_cli.chat.invalid_chunk.count';
+const CONTENT_RETRY_COUNT = 'gemini_cli.chat.content_retry.count';
+const CONTENT_RETRY_FAILURE_COUNT =
+  'gemini_cli.chat.content_retry_failure.count';
+const MODEL_ROUTING_LATENCY = 'gemini_cli.model_routing.latency';
+const MODEL_ROUTING_FAILURE_COUNT = 'gemini_cli.model_routing.failure.count';
+const MODEL_SLASH_COMMAND_CALL_COUNT =
+  'gemini_cli.slash_command.model.call_count';
+
+// Performance Monitoring Metrics
+const STARTUP_TIME = 'gemini_cli.startup.duration';
+const MEMORY_USAGE = 'gemini_cli.memory.usage';
+const CPU_USAGE = 'gemini_cli.cpu.usage';
+const TOOL_QUEUE_DEPTH = 'gemini_cli.tool.queue.depth';
+const TOOL_EXECUTION_BREAKDOWN = 'gemini_cli.tool.execution.breakdown';
+const TOKEN_EFFICIENCY = 'gemini_cli.token.efficiency';
+const API_REQUEST_BREAKDOWN = 'gemini_cli.api.request.breakdown';
+const PERFORMANCE_SCORE = 'gemini_cli.performance.score';
+const REGRESSION_DETECTION = 'gemini_cli.performance.regression';
+const REGRESSION_PERCENTAGE_CHANGE =
+  'gemini_cli.performance.regression.percentage_change';
+const BASELINE_COMPARISON = 'gemini_cli.performance.baseline.comparison';
+
+const baseMetricDefinition = {
+  getCommonAttributes: (config: Config): Attributes => ({
+    'session.id': config.getSessionId(),
+  }),
+};
+
+const COUNTER_DEFINITIONS = {
+  [TOOL_CALL_COUNT]: {
+    description: 'Counts tool calls, tagged by function name and success.',
+    valueType: ValueType.INT,
+    assign: (c: Counter) => (toolCallCounter = c),
+    attributes: {} as {
+      function_name: string;
+      success: boolean;
+      decision?: 'accept' | 'reject' | 'modify' | 'auto_accept';
+      tool_type?: 'native' | 'mcp';
+    },
+  },
+  [API_REQUEST_COUNT]: {
+    description: 'Counts API requests, tagged by model and status.',
+    valueType: ValueType.INT,
+    assign: (c: Counter) => (apiRequestCounter = c),
+    attributes: {} as {
+      model: string;
+      status_code?: number | string;
```

Comment:
Personally, I'd probably hoist a specific type for `status_code` and use either:

- a number (like http status)
- an enum of fixed options (success, failure, etc)
- (least preferred) a string union of success|failure|etc